browser/ui/startup/default_brave_browser_prompt.cc

```diff
@@ -0,0 +1,167 @@
+    // counting on the early return here. See bug crbug.com/693292.
+    if (first_run::IsOnWelcomePage(web_contents))
+      continue;
+
+    // Launch dialog.
+    brave::ShowDefaultBrowserDialog(browser);
+    break;
+  }
+}
+
+void ResetCheckDefaultBrowserPref(const base::FilePath& profile_path) {
+  Profile* profile =
+      g_browser_process->profile_manager()->GetProfileByPath(profile_path);
+  if (profile)
+    ResetDefaultBraveBrowserPrompt(profile);
+}
+
+void OnCheckIsDefaultBrowserFinished(
+    const base::FilePath& profile_path,
+    bool show_prompt,
+    shell_integration::DefaultWebClientState state) {
+  if (state == shell_integration::IS_DEFAULT) {
+    // Notify the user in the future if Chrome ceases to be the user's chosen
+    // default browser.
+    ResetCheckDefaultBrowserPref(profile_path);
+  } else if (show_prompt && state == shell_integration::NOT_DEFAULT &&
+             shell_integration::CanSetAsDefaultBrowser()) {
+    // Only show the prompt if some other program is the user's default browser.
+    // In particular, don't show it if another install mode is default (e.g.,
+    // don't prompt for Chrome Beta if stable Chrome is the default).
+    ShowPrompt();
+  }
+}
+
+// Returns true if the default browser prompt should be shown if Chrome is not
+// the user's default browser.
+bool ShouldShowDefaultBrowserPrompt(Profile* profile) {
+  // Do not show the prompt if "suppress_default_browser_prompt_for_version" in
+  // the initial preferences is set to the current version.
+  const std::string disable_version_string =
+      g_browser_process->local_state()->GetString(
+          prefs::kBrowserSuppressDefaultBrowserPrompt);
+  const base::Version disable_version(disable_version_string);
+  DCHECK(disable_version_string.empty() || disable_version.IsValid());
+  if (disable_version.IsValid() &&
+      disable_version == version_info::GetVersion()) {
+    return false;
+  }
+
+  const int current_count =
+      profile->GetPrefs()->GetInteger(kDefaultBrowserLaunchingCount);
+
+  // We only show prompt at 2nd, 3rd, 4th, 8th and 20th.
+  // This is not called at first run. So, count 1 is second run.
+  if (current_count == 1 || current_count == 2 || current_count == 3 ||
+      current_count == 7 || current_count == 19)
+    return true;
+
+  return false;
+}
+
+}  // namespace
+
+void ShowDefaultBraveBrowserPrompt(Profile* profile) {
+  // Do not check if Chrome is the default browser if there is a policy in
+  // control of this setting.
+  if (g_browser_process->local_state()->IsManagedPreference(
+          prefs::kDefaultBrowserSettingEnabled)) {
+    // Handling of the browser.default_browser_setting_enabled policy setting is
+    // taken care of in BrowserProcessImpl.
+    return;
+  }
+
+  PrefService* prefs = profile->GetPrefs();
+  // Reset preferences if kResetCheckDefaultBrowser is true.
+  if (prefs->GetBoolean(prefs::kResetCheckDefaultBrowser)) {
```

Comment:
That's interesting how this is initialized
```
#if defined(OS_WIN)
  reset_check_default = base::win::GetVersion() >= base::win::Version::WIN10;
#endif
  registry->RegisterBooleanPref(prefs::kResetCheckDefaultBrowser,
                                reset_check_default);
```
not sure whether we need this behavior

Reply:
This would be helpful when win version is updated and default browser setting is changed.
Then, we want to make Brave as a default again.